src/drc/api/filters.py

```diff
@@ -1,10 +1,45 @@
-from vng_api_common.filtersets import FilterSet
+from copy import deepcopy
+from urllib.parse import urlparse
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from django_filters import filters
+from vng_api_common.filters import URLModelChoiceField, URLModelChoiceFilter
+from vng_api_common.filtersets import FILTER_FOR_DBFIELD_DEFAULTS, FilterSet
+from vng_api_common.utils import get_resource_for_path
 
 from drc.datamodel.models import (
-    EnkelvoudigInformatieObject, Gebruiksrechten, ObjectInformatieObject
+    EnkelvoudigInformatieObject, EnkelvoudigInformatieObjectCanonical,
+    Gebruiksrechten, ObjectInformatieObject
 )
 
 
+class InformatieObjectURLChoiceField(URLModelChoiceField):
```

Comment:
Looking at the changes of this subclass compared to the base class, I think it'd make more sense to support an optional parameter `instance_path` which is a python dotted path.

In `vng_api_common.filters.URLModelChoiceField.url_to_pk` you'd then change:

```python
instance = get_resource_for_path(path)
```

into something like:

```python
instance = get_resource_for_path(path)
if self.instance_path:
    for bit in self.instance_path.split('.'):
        instance = getattr(instance, bit)
```